File: src/main/java/com/github/davinkevin/transmissionrss/transmission/request/AddTorrentRequest.java
package com.github.davinkevin.transmissionrss.transmission.request;

import com.github.davinkevin.transmissionrss.transmission.arguments.AddTorrentArguments;

/**
 * Created by kevin on 08/04/2018
 */
public class AddTorrentRequest<T extends AddTorrentArguments> extends RpcRequest<T> {

    private static final String TORRENT_ADD = "torrent-add";

    public AddTorrentRequest(T arguments) {
        super(TORRENT_ADD, arguments, "");
    }
}

File: src/main/kotlin/com/github/davinkevin/transmissionrss/transmission/request/AddTorrentRequest.kt
package com.github.davinkevin.transmissionrss.transmission.request

import com.github.davinkevin.transmissionrss.transmission.arguments.AddTorrentArguments

data class AddTorrentRequest(val arguments: AddTorrentArguments) {
    private val TORRENT_ADD = "torrent-add"

    fun toRpcRequest() = RpcRequest(
            method = TORRENT_ADD,
            arguments = arguments,
            tag = ""
    )
}